src/features/member/sagas/promoteDelegatedAccount.saga.ts

```diff
@@ -1,14 +1,42 @@
 import { MemberAction } from '$/features/member'
+import { TokenGatedRoomAction } from '$/features/tokenGatedRooms'
+import { Address, EnhancedStream, PrivacySetting } from '$/types'
 import handleError from '$/utils/handleError'
 import setMultiplePermissions from '$/utils/setMultiplePermissions'
 import takeEveryUnique from '$/utils/takeEveryUnique'
 import { error, success } from '$/utils/toaster'
+import { put } from 'redux-saga/effects'
 import { StreamPermission } from 'streamr-client'
 
 function* onPromoteDelegatedAccountAction({
-    payload: { roomId, delegatedAddress, provider, requester, streamrClient },
+    payload: { roomId, delegatedAddress, provider, requester, streamrClient, privacy },
 }: ReturnType<typeof MemberAction.promoteDelegatedAccount>) {
     try {
+        // split for token-gated rooms
+        if (privacy === PrivacySetting.TokenGated) {
+            console.log('promoting token gated room')
+            const stream: EnhancedStream = yield streamrClient.getStream(roomId)
+            const metadata = stream.extensions['thechat.eth']
+
+            if (!metadata.tokenAddress) {
+                throw new Error('No token address found')
+            }
+            console.log({ metadata })
+
+            const owner: Address = yield streamrClient.getAddress()
```

Comment:
Value of `streamrClient.getAddress()` should be equal to the value passed as `requester`. Maybe `requester` is redundant… I like the `getAddress` way. Sounds like a future "todo".